### PR TITLE
Add typing

### PR DESCRIPTION
**Pull request by tekktrik**
I believe I typed this correctly.  Was a little unsure of using the CircuitPython "_typing" library.

### CODE REVIEW

**Comment by tekktrik** [comment]
Addressed issues from pre-commit check

File: adafruit_ht16k33/ht16k33.py [hunk 1]
```diff
@@ -13,6 +13,12 @@
 
 from adafruit_bus_device import i2c_device
 from micropython import const
+from busio import I2C
+
+try:
+    from typing import Optional
```
Comment (gamblor21) [inline adafruit_ht16k33/ht16k33.py:18]: You will have to also import ReadableBuffer

File: adafruit_ht16k33/ht16k33.py [hunk 2]
```diff
@@ -45,7 +57,7 @@ def __init__(self, i2c, address=0x70, auto_write=True, brightness=1.0):
         self.blink_rate = 0
         self.brightness = brightness
 
-    def _write_cmd(self, byte):
+    def _write_cmd(self, byte: "_typing.ReadableBuffer"):
```
Comment (gamblor21) [inline adafruit_ht16k33/ht16k33.py]: I believe this can just be `byte: ReadableBuffer` once it is imported.

File: adafruit_ht16k33/matrix.py [hunk 3]
```diff
@@ -205,7 +210,7 @@ def fill(self, color):
         if self._auto_write:
             self.show()
 
-    def image(self, img):
+    def image(self, img: "PIL.Image"):
```
Comment (gamblor21) [inline adafruit_ht16k33/matrix.py]: I think this has to be imported in the try at the top.

**Review by gamblor21 — CHANGES_REQUESTED** [review]
Good work. I didn't test it on actual hardware and noted a couple things I saw.

File: adafruit_ht16k33/matrix.py (hunk 3 above)
Comment (FoamyGuy) [reply adafruit_ht16k33/matrix.py]: Lets import it like this:
```
from PIL import Image
```
and then in the `img: Image` 

This seems to get the IDE able to follow through to the right class which didn't work for me with `import PIL.Image`

**Comment by tekktrik** [comment]
Thanks @gamblor21 and @FoamyGuy for the helpful feedback!  Appreciate the help for a new contributor.  Just resubmitted with changes per review comments.

**Comment by tekktrik** [comment]
It actually just kicked back on building the docs, was it because it failed importing '_typing' so it never imported I2C?

**Comment by tekktrik** [comment]
Actually, rereading the documentation, is '_typing' actually a real library. or should I create a Union named ReadableBuffer within the import instead?

**Comment by FoamyGuy** [comment]
> Actually, rereading the documentation, is '_typing' actually a real library. or should I create a Union named ReadableBuffer within the import instead?

Im not familiar with  `_typing` not sure if it's a real module or not, but I don't think it's one that we've used for any of these so far.

I'm not certain where the ReadableBuffer import should come from though.

I'll take a look into this one and see if I can figure out the right way to import it to make the sphinx build happier. But it's likely I won't be able to get to it until tomorrow evening.

**Comment by tekktrik** [comment]
Much appreciated!  Let me know when you determine how to do it!

File: adafruit_ht16k33/ht16k33.py [hunk 4]
```diff
@@ -45,7 +58,7 @@ def __init__(self, i2c, address=0x70, auto_write=True, brightness=1.0):
         self.blink_rate = 0
         self.brightness = brightness
 
-    def _write_cmd(self, byte):
+    def _write_cmd(self, byte: ReadableBuffer):
```
Comment (FoamyGuy) [inline adafruit_ht16k33/ht16k33.py]: I think we can use `bytearray` for the type here. That is what's listed in the docs for `I2CDevice.write()` Which is what this value gets passed to. https://circuitpython.readthedocs.io/projects/busdevice/en/latest/api.html#adafruit_bus_device.i2c_device.I2CDevice.write

**Comment by tekktrik** [comment]
Awesome, changes made!

**Comment by FoamyGuy** [comment]
Try adding `pillow` to the requirements.txt file. I think that should make sphinx recognize the Image object.

**Comment by tekktrik** [comment]
Awesome, thanks for all the help

**Comment by FoamyGuy** [comment]
It wants an extra empty line at the end of requirements.txt as well. (picky CI). Thanks for working through the issues.

**Comment by tekktrik** [comment]
Haha no worries, thanks for all the help!

**Review by FoamyGuy — APPROVED** [review]
Looks good to me. Thank you @tekktrik

**Pull request merged**